### PR TITLE
dynamicarray_clean

### PR DESCRIPTION
**Pull request by burner**
a clean method that removes all elments from the dynamicarray but does not deallocate the internal array

### CODE REVIEW

File: src/containers/dynamicarray.d [hunk 1]
```diff
@@ -150,6 +150,22 @@ struct DynamicArray(T, Allocator = Mallocator, bool supportGC = shouldAddGCRange
...
 
+	/** Deletes all the elements, but does not disallocate the memory
+	of the DynamicArray.
+	*/
+	void clean()
+	{
+		for (size_t i = 0; i < this.l; ++i)
+		{
+			static if (is(T == class))
+				destroy(arr[i]);
```
Comment (Hackerpilot) [inline src/containers/dynamicarray.d]: Won't this cause problems if the array doesn't own the class instance?

File: src/containers/dynamicarray.d [hunk 2]
```diff
@@ -150,6 +150,22 @@ struct DynamicArray(T, Allocator = Mallocator, bool supportGC = shouldAddGCRange
 		}
 	}
 
+	/** Deletes all the elements, but does not disallocate the memory
+	of the DynamicArray.
+	*/
+	void clean()
```
Comment (Hackerpilot) [inline src/containers/dynamicarray.d]: The other containers in this library as well as the containers in Phobos call this `clear`.

File: src/containers/dynamicarray.d (hunk 1 above)
Comment (burner) [inline src/containers/dynamicarray.d]: It will blow up in your face

File: src/containers/dynamicarray.d (hunk 2 above)
Comment (burner) [inline src/containers/dynamicarray.d]: I'll change that

**Comment by burner** [comment]
ideas?

**Comment by Hackerpilot** [comment]
Didn't you have a merge request that added a `clear()` function?

**Comment by Hackerpilot** [comment]
Also, this pull request has merge conflicts.

**Comment by burner** [comment]
I'll check

**Comment by burner** [comment]
ping

**Pull request closed without merging**